libraries/AP_AccelCal/AccelCalibrator.cpp

```diff
@@ -0,0 +1,477 @@
+        return false;
+    } else {
+        return true;
+    }
+}
+
+// interface for LSq estimator to read sample buffer sent after conversion from delta velocity
+// to averaged acc over time
+bool AccelCalibrator::get_sample(uint8_t i, Vector3f& s) const {
+    if (i >= _samples_collected) {
+        return false;
+    }
+    s = _sample_buffer[i].delta_velocity / _sample_buffer[i].delta_time;
+    return true;
+}
+
+// returns truen and sample corrected with diag offdiag parameters as calculated by LSq estimation procedure
+// returns false if no correct parameter exists to be applied along with existing sample without corrections
+bool AccelCalibrator::get_sample_corrected(uint8_t i, Vector3f& s) const {
+    if (_status != ACCEL_CAL_SUCCESS || !get_sample(i,s)) {
+        return false;
+    }
+
+    Matrix3f M (
+        _param_struct.diag.x    , _param_struct.offdiag.x , _param_struct.offdiag.y,
+        _param_struct.offdiag.x , _param_struct.diag.y    , _param_struct.offdiag.z,
+        _param_struct.offdiag.y , _param_struct.offdiag.z , _param_struct.diag.z
+    );
+
+    s = M*(s+_param_struct.offset);
+
+    return true;
+}
+
+// checks if no new sample has been recieved for considerable amount of time
+void AccelCalibrator::check_for_timeout() {
+    static const uint32_t timeout = _conf_sample_time*2*1000 + 500;
+    if (_status == ACCEL_CAL_COLLECTING_SAMPLE && AP_HAL::millis() - _last_samp_frag_collected_ms > timeout) {
+        set_status(ACCEL_CAL_FAILED);
+    }
+}
+
+// returns spherical fit paramteters
+void AccelCalibrator::get_calibration(Vector3f& offset) const {
+    offset = -_param_struct.offset;
+}
+
+// returns axis aligned ellipsoidal fit parameters
+void AccelCalibrator::get_calibration(Vector3f& offset, Vector3f& diag) const {
+    offset = -_param_struct.offset;
+    diag = _param_struct.diag;
+}
+
+// returns generic ellipsoidal fit parameters
+void AccelCalibrator::get_calibration(Vector3f& offset, Vector3f& diag, Vector3f& offdiag) const {
+    offset = -_param_struct.offset;
+    diag = _param_struct.diag;
+    offdiag = _param_struct.offdiag;
+}
+
+/////////////////////////////////////////////////////////////
+////////////////////// PRIVATE METHODS //////////////////////
+/////////////////////////////////////////////////////////////
+
+/*
+  The sample acceptance distance is determined as follows:
+  For any regular polyhedron with triangular faces, the angle theta subtended
+  by two closest points is defined as
+ 
+       theta = arccos(cos(A)/(1-cos(A)))
+ 
+  Where:
+       A = (4pi/F + pi)/3
+  and
+       F = 2V - 4 is the number of faces for the polyhedron in consideration,
+       which depends on the number of vertices V
+ 
+  The above equation was proved after solving for spherical triangular excess
+  and related equations.
+ */
+bool AccelCalibrator::accept_sample(const Vector3f& sample)
+{
+    static const uint16_t faces = 2*_conf_num_samples-4;
```

Comment:
if _conf_num_samples was const then it would be OK I think, then whole expression would be const